mender-convert-modify

```diff
@@ -277,6 +222,61 @@ ${data_part_device}   /data          auto       ${MENDER_DATA_PART_FSTAB_OPTS}
 $fstab
 EOF
 
+log_info "Installing Mender client and related files"
+
+if [ "${MENDER_CLIENT_INSTALL}" = "y" ]; then
+    case "${MENDER_CLIENT_VERSION}" in
+        1.* | 2.* | 3.*)
+            log_info "Installing Mender client version ${MENDER_CLIENT_VERSION}"
+            deb_get_and_install_package mender-client mender-client "${MENDER_CLIENT_VERSION}"
+            mender_client_deb_name="${DEB_NAME}"
+            ;;
+        *)
+            # For 4.x, master and latest(*) install the new packages
+            # At the time of writing, this will be broken for "latest" until mender-client v4 is
+            # released. Having an special handling deemed to complex for the short time benefit.
+            log_info "Installing Mender Auth and Mender Update version ${MENDER_CLIENT_VERSION}"
+            deb_get_and_install_package mender-client mender-auth "${MENDER_CLIENT_VERSION}"
+            deb_get_and_install_package mender-client mender-update "${MENDER_CLIENT_VERSION}"
+            mender_client_deb_name="${DEB_NAME}"
```

Comment:
Implicit value return - also pre-existing. But I wish we returned it from the function instead

Reply:
I agree. But it's not so trivial to fix this, because it relies on the fetching logic. We'd either have to:

1. Repeat the call in order to get the name (expensive, because we have to re-download every package).
2. Restructure the code so that the download happens first, and then this is passed into the installer afterwards when we already have the name. Doable, but quite an intrusive change, since it puts more work on `mender-convert-modify`.
3. Keep it as is, but print the `DEB_NAME` in the function. Error prone, because there is a lot of output from the deb installer as well.

Definitely not unsolvable, but I think I'll let this dragon sleep for today.

Reply:
Yes, not going to force this on you then :)